src/execution/execute.ts

```diff
@@ -1618,6 +1618,10 @@ export const defaultTypeResolver: GraphQLTypeResolver<unknown, unknown> =
         if (isPromise(isTypeOfResult)) {
           promisedIsTypeOfResults[i] = isTypeOfResult;
         } else if (isTypeOfResult) {
+          if (promisedIsTypeOfResults.length > 0) {
```

Comment:
I think this should probably silently ignore the errors....

Something like:

```ts
if (promisedIsTypeOfResults.length > 0) {
  Promise.all(promisedIsTypeOfResults).then(undefined, () => {
    /* ignore errors */
  });
}
```

Reply:
Seeing as one functioning isTypeOf resolver is all one should have and all one needs.

Compare also to a sync case with multiple sync isTypeOf resolvers, one that returns true, and multiple that throw instead of returning false. If the order is serendipitous, the non-erroring one will return true and mask all the other potential errors.

In #3706, we convert a sync error to an async error, but that's the not happy path of an error. Here, this change will convert every synchronous isTypeOf to a non-synchronous....

Reply:
Compare also to multiple async errors, when we return immediately with the first async error, without awaiting the others. I would argue that #3706 is a bit funny, in that if we have a sync error, there is no need for:

```ts
return promiseForObject(...).finally(() => throw originalError);
```

we should just have:
```ts
promiseForObject(....).then(undefined, () =>{ /*ignore*/ });
throw originalError;
```

Reply:
makes sense, updated

Reply:
Looks good!